### PR TITLE
kustomize deploy

### PR DESCRIPTION
**Pull request by warjiang**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
add kustomize style of deploy, support mode of nodePort and ingress

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
- support deploy mode of nodePort and ingress
```

### CODE REVIEW

File: artifacts/dashboard/karmada-dashboard-api.yaml [hunk 1]
```diff
@@ -32,8 +32,11 @@ spec:
             - --insecure-bind-address=0.0.0.0
             - --bind-address=0.0.0.0
           name: karmada-dashboard-api
-          image: karmada/karmada-dashboard-api:refactor-dashboard
+          image: karmada/karmada-dashboard-api:main
```
Comment (RainbowMango) [inline artifacts/dashboard/karmada-dashboard-api.yaml:35]: I fixed this at #103. You might need to rebase after that.

File: artifacts/overlays/nodeport-mode/dashboard-config.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  - path: /addon
+    enable: false
+    sidebar_key: ADDON
+    children:
+      - path: buildin
+        enable: true
+        sidebar_key: BUILDIN
+      - path: thirdparty
+        enable: true
+        sidebar_key: THIRDPARTY
```
Comment (RainbowMango) [inline artifacts/overlays/nodeport-mode/dashboard-config.yaml:79]: no end of line

File: artifacts/overlays/ingress-mode/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  namespace: karmada-system
+spec:
+  rules:
+    - http:
+        paths:
+          - backend:
+              serviceName: karmada-dashboard-web
+              servicePort: 8000
+            path: /
+            pathType: ImplementationSpecific
```
Comment (RainbowMango) [inline artifacts/overlays/ingress-mode/ingress.yaml:14]: no end of line

File: artifacts/overlays/ingress-mode/dashboard-config.yaml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  - path: /addon
+    enable: false
+    sidebar_key: ADDON
+    children:
+      - path: buildin
+        enable: true
+        sidebar_key: BUILDIN
+      - path: thirdparty
+        enable: true
+        sidebar_key: THIRDPARTY
```
Comment (RainbowMango) [inline artifacts/overlays/ingress-mode/dashboard-config.yaml:79]: no end of line.

File: artifacts/overlays/ingress-mode/.gitignore
Comment (RainbowMango) [inline artifacts/overlays/ingress-mode/.gitignore:1]: Why the .gitignore live in this path?

Comment (warjiang) [reply artifacts/overlays/ingress-mode/.gitignore:1]: I will merge this `.gitignore` file. The reason is that for convinient 😆

**Comment by warjiang** [comment]
@RainbowMango The branch already be rebased, and related problem has been resolved.

File: artifacts/dashboard/karmada-dashboard-api.yaml [hunk 5]
```diff
@@ -34,6 +34,9 @@ spec:
           name: karmada-dashboard-api
           image: karmada/karmada-dashboard-api:main
           imagePullPolicy: IfNotPresent
+          env:
+            - name: GIN_MODE
+              value: release
```
Comment (RainbowMango) [inline artifacts/dashboard/karmada-dashboard-api.yaml:39]: What's the environment used for? I searched over the code base but didn't find any reference to it.

File: artifacts/dashboard/karmada-dashboard-web.yaml [hunk 6]
```diff
@@ -58,12 +62,19 @@ spec:
             - name: kubeconfig
               subPath: kubeconfig
               mountPath: /etc/kubeconfig
+            - mountPath: /config/dashboard-config.yaml
+              name: dashboard-config
+              subPath: prod.yaml
```
Comment (RainbowMango) [inline artifacts/dashboard/karmada-dashboard-web.yaml:67]:
```suggestion
            - name: dashboard-config
              subPath: prod.yaml
              mountPath: /config/dashboard-config.yaml
```

File: artifacts/overlays/ingress-mode/.gitignore [hunk 7]
```diff
@@ -0,0 +1 @@
+.dockerconfigjson
```
Comment (RainbowMango) [inline artifacts/overlays/ingress-mode/.gitignore:1]: no end of line

File: artifacts/dashboard/karmada-dashboard-api.yaml (hunk 5 above)
Comment (warjiang) [reply artifacts/dashboard/karmada-dashboard-api.yaml:39]: It's the env for gin, more details can refer： https://github.com/gin-gonic/gin/blob/master/mode.go

File: artifacts/overlays/ingress-mode/.gitignore (hunk 7 above)
Comment (warjiang) [reply artifacts/overlays/ingress-mode/.gitignore]: I merge the inner .gitignore to the outer one

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**